src/run.jl

```diff
@@ -117,25 +117,27 @@ function active_learn(experiment::Dict{Symbol, Any}, data::Array{T, 2}, labels::
             end
             debug(LOGGER, "[QS] Starting query strategy on $(format_observations(query_data)) observations.")
             if i == 0
-                query, time_qs, mem_qs = @timed get_query_object_helper(qs, query_data, query_pools, query_indices)
+                queries, time_qs, mem_qs = @timed get_query_objects_helper(qs, query_data, query_pools, query_indices)
             else
-                query, time_qs, mem_qs = @timed get_query_object_helper(qs, query_data, query_pools, query_indices, values(res.al_history, :query_history))
+                queries, time_qs, mem_qs = @timed get_query_objects_helper(qs, query_data, query_pools, query_indices, values(res.al_history, :query_history))
             end
-            debug(LOGGER, "[QS] Query strategy finished ($(time_qs) s, $(format_bytes(mem_qs))).")
-            query_label = ask_oracle(oracle, query)
 
-            # tmp workaround
-            data, pools, labels = process_query!(isa(query, Array) ? query : [query],
-                                         isa(query_label, Array) ? query_label : [query_label],
+            debug(LOGGER, "[QS] Query strategy scoring done ($(time_qs) s, $(format_bytes(mem_qs))).")
+            # qs_plot = Visualization.plot_query_heatmap(model, qs)
+            # Visualization.save_plot(qs_plot, "training/qs_score_iteration_$(i)")
+            debug(LOGGER, "[QS] Updating pools.")
+            query_labels = ask_oracle(oracle, queries)
+            data, pools, labels = process_query!(isa(queries, Array) ? queries : [queries],
+                                         isa(query_labels, Array) ? query_labels : [query_labels],
```

Comment:
Same here. The query synthesis oracles are not yet refactored in this regard though. I will look into this myself.